docs/REST API/rest-api.md

```diff
@@ -9138,6 +9141,125 @@ Add App Store (VPP) app purchased in Apple Business Manager.
 
 `Status: 200`
 
+### List Fleet library apps
+
+> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows.
+
+List available Fleet library apps.
+
+`GET /api/v1/fleet/software/fleet_library_apps`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| team_id       | integer | query | **Required**. The team ID. Filters Fleet library apps to only include apps available for the specified team.  |
+
+#### Example
+
+`GET /api/v1/fleet/software/fleet_library_apps?team_id=3`
+
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "fleet_library_apps": [
+    {
+      "id": "1",
+      "name": "1Password",
+      "version": "8.10.40",
+      "platform": "darwin"
+    },
+    {
+      "id": "2",
+      "name": "Adobe Acrobat Reader",
+      "version": "24.002.21005",
+      "platform": "darwin"
+    },
+    {
+      "id": "3",
+      "name": "Box Drive",
+      "version": "2.39.179",
+      "platform": "darwin"
+    },
+  ]
+}
+```
+
+### Get Fleet library app
+
+> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows.
+
+Returns information about the specified Fleet library app.
+
+`GET /api/v1/fleet/software/fleet_library_apps/:id`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| id   | integer | path | **Required.** The Fleet library app's ID. |
+
+
+#### Example
+
+`GET /api/v1/fleet/software/fleet_library_apps/1`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "fleet_library_app": {
+    "id": 1,
+    "name": "1Password",
+    "file_name": "1Password-8.10.44-aarch64.zip",
+    "version": "8.10.40",
+    "platform": "darwin",
+    "install_script": "#!/bin/sh\ninstaller -pkg \"$INSTALLER_PATH\" -target /",
+    "uninstall_script": "#!/bin/sh\npkg_ids=$PACKAGE_ID\nfor pkg_id in '${pkg_ids[@]}'...",
+  }
+}
+```
+
+### Add Fleet library app
+
+> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows.
+
+_Available in Fleet Premium._
+
+Add app from Fleet library.
+
+`POST /api/v1/fleet/software/fleet_library_apps`
```

Comment:
I don't think we need this new endpoint. Just add `library_id` to the current add software endpoint and make the binary OR library_id required. Less to remake

Reply:
Because we are still giving the user the ability to modify the scripts and whatnot so those need to be sent (makes sense to use the current ep)

Reply:
On design review, we decided to keep the endpoint separate because they are seperate UI/UX flows from the user's perspective (different frontend URLs).